DotNet/Report/Application/Factory/CustomMongoSchedulerFactory.cs

```diff
@@ -0,0 +1,171 @@
+using LantanaGroup.Link.Shared.Application.Models.Configs;
+using LantanaGroup.Link.Report.Jobs.JobStoreFactories;
+using Quartz;
+using Quartz.Spi;
+using Reddoxx.Quartz.MongoDbJobStore;
+using Quartz.Simpl;
+using Quartz.Impl;
+
+namespace LantanaGroup.Link.Report.Application.Factory;
+
+public class CustomMongoSchedulerFactory : ISchedulerFactory
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<CustomMongoSchedulerFactory> _logger;
+    private IScheduler? _scheduler;
+    private readonly SemaphoreSlim _lock = new SemaphoreSlim(1, 1);
+
+    public CustomMongoSchedulerFactory(IServiceProvider serviceProvider, ILogger<CustomMongoSchedulerFactory> logger)
+    {
+        _serviceProvider = serviceProvider;
+        _logger = logger;
+    }
+
+    public async Task<IScheduler> GetScheduler(CancellationToken cancellationToken = default)
+    {
+        if (_scheduler != null)
+            return _scheduler;
+
+        await _lock.WaitAsync(cancellationToken);
+        try
+        {
+            if (_scheduler != null)
+                return _scheduler;
+
+            _logger.LogInformation("Creating MongoDB scheduler...");
+
+            var loggerFactory = _serviceProvider.GetRequiredService<ILoggerFactory>();
+            var mongoOptions = _serviceProvider.GetRequiredService<IOptions<MongoConnection>>();
+
+            // Register types for BSON serialization
+            try
+            {
+                if (!MongoDB.Bson.Serialization.BsonClassMap.IsClassMapRegistered(typeof(LantanaGroup.Link.Shared.Application.Models.RetryEntity)))
+                {
+                    MongoDB.Bson.Serialization.BsonClassMap.RegisterClassMap<LantanaGroup.Link.Shared.Application.Models.RetryEntity>(cm =>
+                    {
+                        cm.AutoMap();
+                        cm.SetIgnoreExtraElements(true);
+                    });
+                    _logger.LogInformation("Registered RetryEntity for BSON serialization");
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "RetryEntity may already be registered for BSON serialization");
+            }
+
+            var quartzFactory = new ReportQuartzMongoDbJobStoreFactory(mongoOptions);
+
+            // Create the MongoDbJobStore
+            var mongoJobStore = new MongoDbJobStore(
+                loggerFactory,
+                quartzFactory,
+                _serviceProvider
+            );
+
+
+            // Set properties
+            mongoJobStore.CollectionPrefix = "reportjobs";
+            mongoJobStore.Clustered = true;
+            mongoJobStore.ClusterCheckinInterval = TimeSpan.FromMilliseconds(7500);
+            mongoJobStore.ClusterCheckinMisfireThreshold = TimeSpan.FromMilliseconds(7500);
+
+            // Create thread pool
+            var threadPool = new DefaultThreadPool();
+            threadPool.MaxConcurrency = 5;
```

Comment:
Configurable via appSettings? Would we ever need this to be a value other than 5?